heron/schedulers/src/java/org/apache/heron/scheduler/kubernetes/V1Controller.java

```diff
@@ -415,17 +415,21 @@ private V1StatefulSet createStatefulSet(Resource containerResource, int numberOf
     final String topologyName = getTopologyName();
     final Config runtimeConfiguration = getRuntimeConfiguration();
 
-    // Get and then create Persistent Volume Claims from the CLI.
-    final Map<String, Map<KubernetesConstants.VolumeClaimTemplateConfigKeys, String>> configsPVC =
+    final List<V1Volume> volumes = new LinkedList<>();
```

Comment:
Only a question out of curiosity. Why did you choose the LinkedList implementation?

Reply:
Thank you for taking the time to review this!

This is a good question. The access patterns on the data structure are such that we only tail append, and when we do access elements we iterate across the entire list from beginning to end. We do not require index-based random access and as such the risk of having to resize the Array backing an `ArrayList` is unnecessary.

With our access patterns, appending to `LinkedList` is O(1) and iteration is O(n) whilst with an `ArrayList` appending would be an amortized O(1) and iteration will be O(n).